### PR TITLE
ML-054/improvements

### PR DESCRIPTION
**Pull request by WhatWasThatt**
# Description

update services with modules
update serializers with missing fields
update services with correct options
update services with sql injections
avoid n+1 problem in services
update controllers

## Trello card link

https://trello.com/c/gMAbzqdS

# Checklist:

- [x] My code fixes the problem or adds required functionality
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] I have added tests that prove that my feature works
- [x] New and existing unit tests pass locally with my changes

# Additional requirements:

- [ ] Does your code require any migrations to be ran?
- [ ] Does your code require installing any new gems or system dependencies?
- [ ] Does your code require any changes to current deployment process?

# Additional Information

### CODE REVIEW

File: app/services/users/guest.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+      super(name, *args, &block)
+    end
+
+    private
+
+    def respond_to_missing?(name, include_private)
+      return true if name.to_s.start_with?('can')
+
+      super(name, include_private)
+    end
```
Comment (twill11) [inline app/services/users/guest.rb:19]: remove from private, please

Comment (WhatWasThatt) [reply app/services/users/guest.rb:19]: why?

Comment (twill11) [reply app/services/users/guest.rb:19]: I think we can break the inheritence under the hood, but I'm not sure, so don't decide at your own peril and risk

**Review by twill11 — APPROVED** [review]
thanks for your change! gj

File: app/controllers/api/v1/awards_controller.rb [hunk 2]
```diff
@@ -6,7 +6,7 @@ class AwardsController < ApplicationController
       after_action :verify_authorized
 
       def index
-        @awards = Dictionary::FindService.call(Award, params)
+        @awards = Awards::FindService.call(params)
```
Comment (dmitrymalllpr) [inline app/controllers/api/v1/awards_controller.rb:9]: permit params please

File: app/services/users/guest.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
+module Users
+  class Guest
+    GUEST_USER_PERMISSIONS = %w[can_read_movie? can_read_person?].freeze
+
+    def method_missing(name, *args, &block)
+      return true if GUEST_USER_PERMISSIONS.include?(name.to_s)
+
+      return false if name.to_s.start_with?('can')
+
+      super(name, *args, &block)
```
Comment (dmitrymalllpr) [inline app/services/users/guest.rb:10]: All are good?

File: app/services/users/guest.rb [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    end
+
+    def respond_to_missing?(name, include_private)
+      return true if name.to_s.start_with?('can')
+
+      super(name, include_private)
```
Comment (dmitrymalllpr) [inline app/services/users/guest.rb:16]: Are you sure?

File: app/services/view_lists/find_service.rb [hunk 5]
```diff
@@ -1,8 +1,8 @@
-module EntityManager
-  class ViewListService < ApplicationService
+module ViewLists
+  class FindService < ApplicationService
     def initialize(params = {})
       @user = params[:user]
-      @params = params[:attributes]
+      @params = params
```
Comment (dmitrymalllpr) [inline app/services/view_lists/find_service.rb:5]: rework please

**Review by dmitrymalllpr — CHANGES_REQUESTED** [review]

File: app/services/users/guest.rb (hunk 3 above)
Comment (dmitrymalllpr) [reply app/services/users/guest.rb:10]: okay

**Review by twill11 — APPROVED** [review]
nice work!

File: app/services/awards/find_service.rb [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+    def initialize(options = { filter: nil, page: nil, order: nil })
+      @filter = options[:filter]
+      @page = options[:page]
+      @order = options[:order]
+    end
+
+    def call
+      if filter.present?
+        Award.includes(%i[categories country])
+             .where('name ILIKE ?', "%#{filter}%")
```
Comment (dmitrymalllpr) [inline app/services/awards/find_service.rb:12]: https://stackoverflow.com/questions/25416667/rails-4-prevent-sql-injection-using-like-operator-in-sql-query
https://rorsecurity.info/portfolio/rails-sql-injection-like

File: app/services/categories/find_service.rb [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+    def initialize(options = { filter: nil, page: nil, order: nil, award_id: nil })
+      @filter = options[:filter]
+      @page = options[:page]
+      @order = options[:order]
+      @award_id = options[:award_id]
+    end
+
+    def call
+      if filter.present?
+        Category.where('award_id = ? AND name ILIKE ?', award_id, "%#{filter}%")
```
Comment (dmitrymalllpr) [inline app/services/categories/find_service.rb:12]: same

File: app/services/dictionary/find_service.rb [hunk 8]
```diff
@@ -1,24 +1,27 @@
...
+      @page = options[:page]
+      @order = options[:order]
+      @name = options[:name]
     end
 
     def call
-      if params[:filter].present?
-        class_name.where('name ILIKE ?', "%#{params[:filter]}%").paginate(page: params[:page]).order("name #{order}")
+      if filter.present?
+        class_name.where('name ILIKE ?', "%#{filter}%").paginate(page: page).order("name #{order}")
```
Comment (dmitrymalllpr) [inline app/services/dictionary/find_service.rb:13]: same

**Review by dmitrymalllpr — CHANGES_REQUESTED** [review]

**Review by twill11 — APPROVED** [review]
gj!

**Review by dmitrymalllpr — APPROVED** [review]

**Pull request merged**